src/errors.rs

```diff
@@ -15,7 +16,7 @@ pub enum BodyErrorCause {
 #[derive(Debug, Clone)]
 pub struct BodyError {
     pub detail: String,
-    pub cause: BodyErrorCause
+    pub cause: sync::Arc<BodyErrorCause>
```

Comment:
Why do we need this to be Clone? The user can always wrap an instance in `Arc` if they need to clone it.